src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -1307,36 +1297,46 @@ public void updateApplicationMetadata(UUID guid, org.cloudfoundry.client.v3.Meta
     @Override
     public void updateApplicationStaging(String applicationName, Staging staging) {
         UUID applicationGuid = getRequiredApplicationGuid(applicationName);
-        UpdateApplicationRequest updateApplicationRequest = UpdateApplicationRequest.builder()
-                                                                                    .applicationId(applicationGuid.toString())
-                                                                                    .lifecycle(buildApplicationLifecycle(staging))
-                                                                                    .build();
         delegate.applicationsV3()
-                .update(updateApplicationRequest)
+                .update(UpdateApplicationRequest.builder()
+                                                .applicationId(applicationGuid.toString())
+                                                .lifecycle(buildApplicationLifecycle(staging))
+                                                .build())
                 .block();
         GetApplicationProcessResponse applicationProcess = getApplicationProcess(applicationGuid);
         updateApplicationProcess(applicationGuid, staging, applicationProcess);
     }
 
     @Override
     public void updateApplicationRoutes(String applicationName, Set<CloudRouteSummary> updatedRoutes) {
-        CloudApplication application = getApplication(applicationName);
-        Set<CloudRouteSummary> currentRoutes = application.getRoutes();
+        UUID applicationGuid = getApplicationGuid(applicationName);
+        List<RouteResource> appRoutes = getRouteResourcesByAppGuid(applicationGuid).collectList()
+                                                                                   .block();
+
+        List<RouteResource> outdatedRoutes = filterRoutes(appRoutes, updatedRoutes);
```

Comment:
In my opinion, RouteResource should be mapped to CloudRoute / CloudRouteSummary and then the same method for filtration should be used just with swapped arguments

Reply:
`CloudRoute` contains the `CloudDomain` object in it and although it is nullable, `RawCloudRoute` _needs_ the domain in order to build the route object.
The way it's done here is to skip the extra request to fetch the domain.